nautobot_topology_views/views.py

```
@@ -349,71 +349,51 @@ def get_topology_data(
 
         if show_logical_connections:
             path_complete_interfaces = Interface.objects.filter(
-                Q(_path__is_complete=True) & Q(device_id__in=device_ids)
+                Q(_path__is_active=True) & Q(device_id__in=device_ids)
             )
             for path_complete_interface in path_complete_interfaces:
-                for connected_endpoint in path_complete_interface.connected_endpoints:
-                    if type(connected_endpoint) != ProviderNetwork:
-                        device_ids.append(connected_endpoint.device.id)
+                connected = path_complete_interface.connected_endpoint
+                if connected is not None and type(connected) != ProviderNetwork:
```

Ruff (0.15.6)

[error] Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)